core/commands/add.go

```diff
@@ -233,9 +227,7 @@ You can now refer to the added file in a gateway, like so:
 		}
 
 		var showProgressBar bool
-		if prgFound {
-			showProgressBar = progress
-		} else if !quiet && !silent {
+		if !progress && !quiet && !silent {
```

Comment:
i'm not sure this one is correct. *If* the user actually specifies a choice for the progress option, we want to respect it, otherwise, we assume true unless they pass either quiet or silent.